pool/upgrades.go

```diff
@@ -136,6 +148,181 @@ func transactionIDUpgrade(tx *bolt.Tx) error {
 	return setDBVersion(tx, newVersion)
 }
 
+func shareIDUpgrade(tx *bolt.Tx) error {
+	const oldVersion = 1
+	const newVersion = 2
+
+	dbVersion, err := fetchDBVersion(tx)
+	if err != nil {
+		return err
+	}
+
+	if dbVersion != oldVersion {
+		desc := "shareIDUpgrade inappropriately called"
+		return MakeError(ErrDBUpgrade, desc, nil)
+	}
+
+	pbkt := tx.Bucket(poolBkt)
+	if pbkt == nil {
+		desc := fmt.Sprintf("bucket %s not found", string(poolBkt))
+		return MakeError(ErrBucketNotFound, desc, nil)
+	}
+
+	sbkt := pbkt.Bucket(shareBkt)
+	if sbkt == nil {
+		desc := fmt.Sprintf("bucket %s not found", string(shareBkt))
+		return MakeError(ErrBucketNotFound, desc, nil)
+	}
+
+	id := func(account string, createdOn int64) string {
```

Comment:
Why is the ID function duplicated here rather than using the function from `share.go`?

In the payment upgrade, the ID function from `payment.go` is used.

Reply:
Yeah, I missed a step there. The `shareID` should be taking the `createdOn` as a parameter, it'll be usable in both cases as a result.